lib/openhab/core/items/persistence.rb

```diff
@@ -255,7 +255,7 @@ def timestamp
         def previous_state(service = nil, skip_equal: false)
           service ||= persistence_service
           result = Actions::PersistenceExtensions.previous_state(self, skip_equal, service&.to_s)
-          HistoricState.new(quantify(result.state), result)
+          HistoricState.new(quantify(result.state), result) if result.is_a?(org.openhab.core.persistence.HistoricItem)
```

Comment:
why not just `if result`, without the type check?

Reply:
can do